### PR TITLE
JS: add xlink:href as xss target when using setAttribute

### PR DESCRIPTION
**Pull request by erik-semmle**
Adds the `xlink:href` attribute to the list of vulnerable attributes.

### CODE REVIEW

File: javascript/ql/test/query-tests/Security/CWE-079/tst3.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ var data = JSON.parse(decodeURIComponent(window.location.search.substr(1)));
 foo.setAttribute("src", data.src); // NOT OK
 foo.setAttribute("HREF", data.p);  // NOT OK
 foo.setAttribute("width", data.w); // OK
+foo.setAttribute("xlink:href", data.p) // NOT OK
```
Comment (xiemaisi) [inline javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: I assume one could also set this attribute using `setAttributeNS`? If so, could you add a test for that?

Comment (erik-semmle) [reply javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: Yes, one could do that. 

The implementation actually detected that correctly before this change, but I added a test for it.

File: javascript/ql/test/query-tests/Security/CWE-079/tst3.js [hunk 2]
```diff
@@ -4,6 +4,11 @@ var data = JSON.parse(decodeURIComponent(window.location.search.substr(1)));
...
+
+foo.setAttributeNS('foobar', 'href', data.p); // NOT OK
```
Comment (xiemaisi) [inline javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: Why "foobar"? Isn't `xlink` from above a very natural example?

Comment (erik-semmle) [reply javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: Yes, `xlink` is a natural example. 
However, I wanted the test to explicitly show that the query matches the `href` attribute with any namespace when using the `setAttributeNS` method. 
(This was the case already before my change in DOM.qll)

Comment (xiemaisi) [reply javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: Then why not have both?

Comment (erik-semmle) [reply javascript/ql/test/query-tests/Security/CWE-079/tst3.js]: I'll do that.

**Pull request merged**